valhalla/jawn/src/lib/stores/ScoreStore.ts

```diff
@@ -83,106 +98,164 @@ export class ScoreStore extends BaseStore {
     }
   }
 
-  public async putScoresIntoClickhouse(newVersion: {
-    id: string;
-    version: number;
-    provider: string;
-    scores: Score[];
-  }): Promise<Result<InsertRequestResponseVersioned, string>> {
+  public async putScoresIntoClickhouse(
+    newVersions: BatchScores[]
+  ): Promise<Result<InsertRequestResponseVersioned[], string>> {
+    const queryPlaceholders = newVersions
+      .map((_, index) => {
+        const base = index * 4;
+        return `({val_${base} : String}, {val_${base + 1} : String}, {val_${
+          base + 2
+        } : String}, {val_${base + 3} : UInt64})`;
+      })
+      .join(",\n    ");
+
+    const queryParams: (string | number | boolean | Date)[] =
+      newVersions.flatMap((v) => [
+        v.requestId,
+        v.organizationId,
+        v.provider,
+        v.version - 1,
+      ]);
+
     let rowContents = resultMap(
       await clickhouseDb.dbQuery<InsertRequestResponseVersioned>(
         `
-      SELECT *
-      FROM request_response_versioned
-      WHERE request_id = {val_0: UUID}
-      AND version = {val_1: UInt64}
-      AND organization_id = {val_2: String}
-      AND provider = {val_3: String}
-    `,
-        [
-          newVersion.id,
-          newVersion.version - 1,
-          this.organizationId,
-          newVersion.provider,
-        ]
+        SELECT *
+        FROM request_response_versioned
+        WHERE (request_id, organization_id, provider, version) IN (${queryPlaceholders})
```

Comment:
I'd be curious to know the performance of doing it like this vs doing something like this

```sql
WHERE (
organization_id IN (...)
  AND provider IN (...)
  AND version IN (...)
  AND request_id IN (...)
)
```